src/main/java/SentinelRefactoring.java

```diff
@@ -0,0 +1,308 @@
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.List;
+
+import org.eclipse.jdt.core.dom.AST;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.Assignment;
+import org.eclipse.jdt.core.dom.Block;
+import org.eclipse.jdt.core.dom.Expression;
+import org.eclipse.jdt.core.dom.ExpressionStatement;
+import org.eclipse.jdt.core.dom.IfStatement;
+import org.eclipse.jdt.core.dom.InfixExpression;
+import org.eclipse.jdt.core.dom.NullLiteral;
+import org.eclipse.jdt.core.dom.NumberLiteral;
+import org.eclipse.jdt.core.dom.PrefixExpression;
+import org.eclipse.jdt.core.dom.SimpleName;
+import org.eclipse.jdt.core.dom.rewrite.ASTRewrite;
+
+import org.eclipse.jdt.core.dom.Statement;
+
+/**
+ * This class represents a refactoring in which integer variables whose values
```

Comment:
Does this work for all kinds of primitive integers, or only `int`? In theory, I don't see why it wouldn't work for `short`, `long`, `byte`, `char`, etc.

What about `Integer` and the other box types?

Reply:
It's okay if these aren't handled (yet?), but please make the Javadoc clear and precise about what will be refactored by the implementation and what will not be.